crates/ruma-events/src/matrix_rtc/member_data.rs

```diff
@@ -0,0 +1,311 @@
+//! Types for matrixRTC `m.call.member` state event content data ([MSC3401])
+//!
+//! [MSC3401]: https://github.com/matrix-org/matrix-spec-proposals/pull/3401
+
+use std::time::Duration;
+
+use as_variant::as_variant;
+use ruma_common::MilliSecondsSinceUnixEpoch;
+use ruma_macros::StringEnum;
+use serde::{Deserialize, Serialize};
+use tracing::warn;
+
+use super::focus::{ActiveFocus, ActiveLivekitFocus, Focus};
+use crate::PrivOwnedStr;
+
+/// The data object that contains the information for one membership.
+///
+/// It can be a legacy or a normal matrixRTC Session membership.
+///
+/// The legacy format contains expired information can cen be expired.
+/// SessionMembershipData does not have the concept of expiration anymore.
+/// It will reliably be removed from the state event when appropriate.
+#[derive(Clone, Debug)]
+#[cfg_attr(test, derive(PartialEq))]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub enum MembershipData<'a> {
+    /// The legacy format (using an array of memberships for each deivece -> one event per user)
+    Legacy(&'a LegacyMembershipData),
+    /// One event per device. `SessionMembershipData` contains all the information required to
+    /// represent the current membership state of one device.
+    Session(&'a SessionMembershipData),
+}
+
+impl<'a> MembershipData<'a> {
+    /// The application this RTC membership participates in (the session type, can be `m.call`...)
+    pub fn get_application(&self) -> &Application {
```

Comment:
[The `get_` prefix is not widely used for accessors in Rust](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter). Please remove them.